### ISSUE TITLE
⚠️ Warn users against gotchas when using object literal property

### ISSUE THREAD

**Issue by TimvdLippe** [Polymer/polymer-linter#172]
_From @samccone on August 10, 2015 20:28_

This bit me today :calendar: , and I think a :warning: warning or :rotating_light: helpful message in the console ala react would help a bunch. :bow: 
#### :goat: Example

``` js
Polymer({
  is: 'goat-eater',
  properties: {
    goats: {
      type: Object,
      value: {
        heads: 1
      }
    }
  },
  attached: function() {
    this.goats.heads++;
    console.log('i have eaten ' + this.goats.heads + ' goat heads');
  } 
});
```

``` html
<goat-eater></goat-eater>
<goat-eater></goat-eater>
<goat-eater></goat-eater>
```

Then in the console I see

```
> i have eaten 1 goat heads
> i have eaten 2 goat heads
> i have eaten 3 goat heads
```

**wat** 

This was pretty darn confusing, until I was like oh yes prototypically inheritance :lollipop: 

``` diff
Polymer({
  is: 'goat-eater',
  properties: {
    goats: {
      type: Object,
-      value: {
+      value: function() {
+       return {
          heads: 1
        }
+     }        
+    }
  }
```

---
### TLDR;

I am willing to bet quite a few people have gotten caught up on using object literal props, it also seems like something that could be `console.warned` against quite easily, given the fact that we have `type` to look at.

Thoughts? Terrible Idea? Let me know :microphone: 

Thanks for your :watch: Time :sparkles: 

_Copied from original issue: Polymer/polymer#2240_

**Comment by TimvdLippe** [comment]
_From @addyosmani on August 10, 2015 21:17_

Afaik We've documented this but agree a warning could be valuable. We've discussed improving warnings/tips around gotchas in the console some time ago but this is definitely an area that could use some more work.

**Comment by TimvdLippe** [comment]
_From @samccone on August 10, 2015 21:23_

:book: https://www.polymer-project.org/1.0/docs/devguide/properties.html#configure-values

This raises a great question, do we want people to ever "use" this feature? Are there legitimate use cases?

**Comment by TimvdLippe** [comment]
_From @sorvell on August 14, 2015 16:43_

@garlicnation This is probably a good candidate for the linter tool.

**Comment by TimvdLippe** [comment]
_From @brettpostin on June 27, 2017 8:30_

+1 for some form of warning/linting. It's very easy to get caught out by this, especially in a componentised world where this is not the obvious behaviour.
